manpage/dotdrop.1
.\" Text automatically generated by txt2man
.TH dotdrop 1 "16 December 2024" "dotdrop-1.15.0" "Save your dotfiles once, deploy them everywhere"
.SH NAME
\fBdotdrop \fP- save your dotfiles once, deploy them everywhere
\fB
.SH SYNOPSIS
.nf
.fam C
\fBdotdrop\fP \fICOMMAND\fP [\fIOPTIONS\fP] [\fIARGS\fP] \.\.\.
\fBdotdrop\fP \fB--help\fP
\fBdotdrop\fP \fB--version\fP

.fam T
.fi
.fam T
.fi
.SH DESCRIPTION
Dotdrop makes the management of dotfiles between different hosts easy.
It allows you to store your dotfiles in Git and automagically deploy
different versions of the same file on different setups.
.PP
It also allows manage different sets of dotfiles. For example, you can have
a set of dotfiles for your home laptop and a different set for your office
desktop. Those sets may overlap, and different versions of the same dotfiles
can be deployed using different predefined profiles. Or you may have a main
set of dotfiles for your everyday host and a subset you only need to deploy
to temporary hosts (cloud VM etc.) that may be using a slightly different
version of some of the dotfiles.
.SH COMMANDS
.TP
.B
install
Install dotfiles
.RS
.TP
.B
\fB-a\fP \fB--force-actions\fP
Execute all actions even if no dotfile is installed.
.TP
.B
\fB-d\fP \fB--dry\fP
Dry run.
.TP
.B
\fB-D\fP \fB--showdiff\fP
Show a diff before overwriting.
.TP
.B
\fB-f\fP \fB--force\fP
Do not ask user confirmation for anything.
.TP
.B
\fB-n\fP \fB--nodiff\fP
Do not diff when installing.
.TP
.B
\fB-p\fP \fB--profile\fP=<profile>
Specify the profile to use.
.TP
.B
\fB-t\fP \fB--temp\fP
Install to a temporary directory for review.
.TP
.B
\fB-w\fP \fB--workers\fP=<nb>
Number of concurrent workers [default: 1].
.TP
.B
\fB-W\fP \fB--workdir-clear\fP
Clear the workdir.
.RE
.TP
.B
import
Import dotfiles
.RS
.TP
.B
\fB-d\fP \fB--dry\fP
Dry run.
.TP
.B
\fB-f\fP \fB--force\fP
Do not ask user confirmation for anything.
.TP
.B
\fB-i\fP \fB--ignore\fP=<pattern>
Pattern to ignore.
.TP
.B
\fB-l\fP \fB--link\fP=<link>
Link option (nolink|absolute|relative|link_children).
.TP
.B
\fB-m\fP \fB--preserve-mode\fP
Insert a chmod entry in the dotfile with its mode.
.TP
.B
\fB-p\fP \fB--profile\fP=<profile>
Specify the profile to use.
.TP
.B
\fB-s\fP \fB--as\fP=<path>
Import as a different path from actual path.
.TP
.B
\fB--transr\fP=<key>
Associate trans_install key on import.
.TP
.B
\fB--transw\fP=<key>
Apply trans_update key on import.
.RE
.TP
.B
compare
Compare dotfiles
.RS
.TP
.B
\fB-C\fP \fB--file\fP=<path>
Path of dotfile to compare.
.TP
.B
\fB-i\fP \fB--ignore\fP=<pattern>
Pattern to ignore.
.TP
.B
\fB-p\fP \fB--profile\fP=<profile>
Specify the profile to use.
.TP
.B
\fB-w\fP \fB--workers\fP=<nb>
Number of concurrent workers [default: 1].
.TP
.B
\fB-z\fP \fB--ignore-missing\fP
Ignore files in installed folders that are missing.
.RE
.TP
.B
update
Update a managed dotfile
.RS
.TP
.B
\fB-d\fP \fB--dry\fP
Dry run.
.TP
.B
\fB-f\fP \fB--force\fP
Do not ask user confirmation for anything.
.TP
.B
\fB-i\fP \fB--ignore\fP=<pattern>
Pattern to ignore.
.TP
.B
\fB-k\fP \fB--key\fP
Treat <path> as a dotfile key.
.TP
.B
\fB-p\fP \fB--profile\fP=<profile>
Specify the profile to use.
.TP
.B
\fB-P\fP \fB--show-patch\fP
Provide a one-liner to manually patch template.
.TP
.B
\fB-w\fP \fB--workers\fP=<nb>
Number of concurrent workers [default: 1].
.TP
.B
\fB-z\fP \fB--ignore-missing\fP
Ignore files in installed folders that are missing.
.RE
.TP
.B
remove
Unmanage a dotfile
.RS
.TP
.B
\fB-d\fP \fB--dry\fP
Dry run.
.TP
.B
\fB-f\fP \fB--force\fP
Do not ask user confirmation for anything.
.TP
.B
\fB-k\fP \fB--key\fP
Treat <path> as a dotfile key.
.TP
.B
\fB-p\fP \fB--profile\fP=<profile>
Specify the profile to use.
.RE
.TP
.B
files
List the managed dotfiles
.RS
.TP
.B
\fB-G\fP \fB--grepable\fP
Grepable output.
.TP
.B
\fB-p\fP \fB--profile\fP=<profile>
Specify the profile to use.
.TP
.B
\fB-T\fP \fB--template\fP
Only template dotfiles.
.RE
.TP
.B
detail
Detail managed dotfiles
.RS
.TP
.B
\fB-p\fP \fB--profile\fP=<profile>
Specify the profile to use.
.RE
.TP
.B
profiles
List all profiles
.RS
.TP
.B
\fB-G\fP \fB--grepable\fP
Grepable output.
.SH GLOBAL OPTIONS
.TP
.B
\fB-b\fP \fB--no-banner\fP
Do not display the banner.
.TP
.B
\fB-c\fP \fB--cfg\fP=<path>
Path to the config.
.TP
.B
\fB-V\fP \fB--verbose\fP
Be verbose.
.SH EXAMPLES
.TP
.B
\fBdotdrop\fP install
[\fB-VbtfndDaW\fP] [\fB-c\fP <path>] [\fB-p\fP <profile>]
[\fB-w\fP <nb>] [<key>\.\.\.]
.TP
.B
\fBdotdrop\fP import
[\fB-Vbdfm\fP]     [\fB-c\fP <path>] [\fB-p\fP <profile>] [\fB-i\fP <pattern>\.\.\.]
[\fB--transr\fP=<key>] [\fB--transw\fP=<key>]
[\fB-l\fP <link>] [\fB-s\fP <path>] <path>\.\.\.
.TP
.B
\fBdotdrop\fP compare
[\fB-LVbz\fP]      [\fB-c\fP <path>] [\fB-p\fP <profile>]
[\fB-w\fP <nb>] [\fB-C\fP <file>\.\.\.] [\fB-i\fP <pattern>\.\.\.]
.TP
.B
\fBdotdrop\fP update
[\fB-VbfdkPz\fP]   [\fB-c\fP <path>] [\fB-p\fP <profile>]
[\fB-w\fP <nb>] [\fB-i\fP <pattern>\.\.\.] [<path>\.\.\.]
.TP
.B
\fBdotdrop\fP remove
[\fB-Vbfdk\fP]     [\fB-c\fP <path>] [\fB-p\fP <profile>] [<path>\.\.\.]
.TP
.B
\fBdotdrop\fP files
[\fB-VbTG\fP]      [\fB-c\fP <path>] [\fB-p\fP <profile>]
.TP
.B
\fBdotdrop\fP detail
[\fB-Vb\fP]        [\fB-c\fP <path>] [\fB-p\fP <profile>] [<key>\.\.\.]
.TP
.B
\fBdotdrop\fP profiles
[\fB-VbG\fP]       [\fB-c\fP <path>]
.PP
\fBdotdrop\fP \fB--help\fP
.PP
\fBdotdrop\fP \fB--version\fP
.SH ABOUT
More information can be found on the repository under <https://github.com/deadc0de6/\fBdotdrop\fP>
and in the main documentation under <https://dotdrop.readthedocs.io>
.SH AUTHOR
Written by deadc0de6 <https://github.com/deadc0de6>
.SH COPYRIGHT
\fBdotdrop\fP is copyright (c) 2017, deadc0de6. Released under the GPLv3 license.
.SH REPORTING BUGS
<https://github.com/deadc0de6/\fBdotdrop\fP/issues>


CHANGELOG.md
# version 0.6

Modified:

  * migrate profiles' dotfiles under new entry "dotfiles" in config

Added:

  * adding "include" to include all dotfiles from another profile
    ([#12](https://github.com/deadc0de6/dotdrop/issues/12))

# version 0.5

Added:

  * handling of *actions* for dotfiles

Fixed:

  * issue with absent dotfile
  * issue with importing symlinks

# version 0.4

Added:

  * new command *listfiles* to list files related to a specific
    profile

# version 0.3

Modified:

  * cleaning usage
  * be quiet by default on *install* and *compare* commands



completion/dotdrop.sh-completion.bash

_dotdropsh()
{
    local cur
    cur="${COMP_WORDS[COMP_CWORD]}"

    if [ $COMP_CWORD -eq 1 ]; then
        COMPREPLY=( $( compgen -W '-h --help -v --version install import compare update remove files detail profiles' -- $cur) )
    else
        case ${COMP_WORDS[1]} in
            install)
            _dotdropsh_install
        ;;
            import)
            _dotdropsh_import
        ;;
            compare)
            _dotdropsh_compare
        ;;
            update)
            _dotdropsh_update
        ;;
            remove)
            _dotdropsh_remove
        ;;
            files)
            _dotdropsh_files
        ;;
            detail)
            _dotdropsh_detail
        ;;
            profiles)
            _dotdropsh_profiles
        ;;
        esac

    fi
}

_dotdropsh_install()
{
    local cur
    cur="${COMP_WORDS[COMP_CWORD]}"

    if [ $COMP_CWORD -ge 2 ]; then
        COMPREPLY=( $( compgen -fW '-V --verbose -b --no-banner -t --temp -f --force -n --nodiff -d --dry -D --showdiff -a --force-actions -c= --cfg= -p= --profile= ' -- $cur) )
    fi
}

_dotdropsh_import()
{
    local cur
    cur="${COMP_WORDS[COMP_CWORD]}"

    if [ $COMP_CWORD -ge 2 ]; then
        COMPREPLY=( $( compgen -fW '-V --verbose -b --no-banner -d --dry -f --force -c= --cfg= -p= --profile= -l= --link= ' -- $cur) )
    fi
}

_dotdropsh_compare()
{
    local cur
    cur="${COMP_WORDS[COMP_CWORD]}"

    if [ $COMP_CWORD -ge 2 ]; then
        COMPREPLY=( $( compgen -W '-V --verbose -b --no-banner -c= --cfg= -p= --profile= -C= --file= -i= --ignore= ' -- $cur) )
    fi
}

_dotdropsh_update()
{
    local cur
    cur="${COMP_WORDS[COMP_CWORD]}"

    if [ $COMP_CWORD -ge 2 ]; then
        COMPREPLY=( $( compgen -fW '-V --verbose -b --no-banner -f --force -d --dry -k --key -P --show-patch -c= --cfg= -p= --profile= -i= --ignore= ' -- $cur) )
    fi
}

_dotdropsh_remove()
{
    local cur
    cur="${COMP_WORDS[COMP_CWORD]}"

    if [ $COMP_CWORD -ge 2 ]; then
        COMPREPLY=( $( compgen -fW '-V --verbose -b --no-banner -f --force -d --dry -k --key -c= --cfg= -p= --profile= ' -- $cur) )
    fi
}

_dotdropsh_files()
{
    local cur
    cur="${COMP_WORDS[COMP_CWORD]}"

    if [ $COMP_CWORD -ge 2 ]; then
        COMPREPLY=( $( compgen -W '-V --verbose -b --no-banner -T --template -G --grepable -c= --cfg= -p= --profile= ' -- $cur) )
    fi
}

_dotdropsh_detail()
{
    local cur
    cur="${COMP_WORDS[COMP_CWORD]}"

    if [ $COMP_CWORD -ge 2 ]; then
        COMPREPLY=( $( compgen -fW '-V --verbose -b --no-banner -c= --cfg= -p= --profile= ' -- $cur) )
    fi
}

_dotdropsh_profiles()
{
    local cur
    cur="${COMP_WORDS[COMP_CWORD]}"

    if [ $COMP_CWORD -ge 2 ]; then
        COMPREPLY=( $( compgen -W '-V --verbose -b --no-banner -G --grepable -c= --cfg= ' -- $cur) )
    fi
}

complete -o bashdefault -o default -o filenames -F _dotdropsh dotdrop.sh